text/0000-default-and-expanded-rustc-errors.md

```diff
@@ -0,0 +1,191 @@
+
+Currently, --explain text focuses on the error code.  You invoke the compiler with --explain <error code> and receive a verbose description of what causes errors of that number.  The resulting message can be helpful, but it uses generic sample code which makes it feel less connected to the user's code.
+
+We propose changing --explain to no longer take an error code.  Instead, passing --explain to the compiler (or to cargo) will turn the compiler output into an expanded form which incorporates the same source and label information the user saw in the default message with more explanation text.
+
+![Image of Rust error in elm-style](http://www.jonathanturner.org/images/elm_like_rust.png)
+
+*Example of an expanded error message*
+
+The expanded error message effectively becomes a template.  The text of the template is the educational text that is explaining the message more more detail.  The template is then populated using the source lines, labels, and spans from the same compiler message that's printed in the default mode.  This lets the message writer call out each label or span as appropriate in the expanded text.
+
+It's possible to also add additional labels that aren't necessarily shown in the default error mode but would be available in the expanded error format.  For example, the above error might look like this is as a default error:
+
+![Image of same error without all of the same labels](http://www.jonathanturner.org/images/default_borrowed_content.png)
+
+This gives the explain text writer maximal flexibility without impacting the readability of the default message.  I'm currently prototyping an implementation of how this templating could work in practice.
+
+## Tying it together
+
+Lastly, we propose that the final error message:
+
+```
+error: aborting due to 2 previous errors
+```
+
+Be changed to notify users of this ability:
+
+```
+note: You can compile again with --explain for more information about these errors
+```
+
+As this helps inform the user of the --explain capability.
+
+# Drawbacks
+
+Changes in the error format can impact integration with other tools.  For example, IDEs that use a simple regex to detect the error would need to be updated to support the new format.  This takes time and community coordination.
+
+While the new error format has a lot of benefits, it's possible that some errors will feel "shoehorned" into it and, even after careful selection of secondary labels, may still not read as well as the original format.  
+
+There is a fair amount of work involved to update the errors and explain text to the proposed format.  
+
+# Alternatives
+
+Rather than using the proposed error format format, we could only provide the verbose --explain style that is proposed in this RFC.  Famous programmers like [John Carmack](https://twitter.com/ID_AA_Carmack/status/735197548034412546) have praised the Elm error format.
+
+![Image of Elm error](http://www.jonathanturner.org/images/elm_error.jpg)
```

Comment:
This image is rather visual trickery. You should make sure that the same colorscheme is used throughout the RFC, to avoid unintended bias.